src/common/AdaptiveStream.cpp

```diff
@@ -90,6 +86,244 @@ void AdaptiveStream::Reset()
   absolutePTSOffset_ = 0;
 }
 
+void adaptive::AdaptiveStream::AllocateSegmentBuffers(size_t size)
+{
+  size++; // TTHR
+
+  while (size-- > 0)
+  {
+    segment_buffers_.emplace_back(new SEGMENTBUFFER());
```

Comment:
Why not use a unique_ptr?

Reply:
because we need higher performance here its what i mentioned in description above
if i use unique_ptr's need double indirections and std::rotate becomes very heavy to process, become even worse than before:
11.177 ms of uniqueptr instead of 0.04552 ms by using raw pointers
although it makes objects creation/destruction easier, it is not worth

Reply:
Have you considered using std::deque instead of std::rotate and std::vector? If all you want is to move the front to the back that may be a better option (and much faster).

Reply:
yes already thought about something similar
but our old code is quite mess and difficult to understand
nor me nor glennguy we are able to understand in full how works all things, so we are changing/improving the code bit by bit, there are still grey areas in the use of buffering variables so atm i dont touch it